### PR TITLE
Add IDEA desktop editor

### PR DESCRIPTION
**Pull request by azatsarynnyy**
⚠️  **Note, this PR depends on merging the following PRs:**
- https://github.com/redhat-developer/devspaces-images/pull/560
- https://gitlab.cee.redhat.com/codeready-workspaces/crw-jenkins/-/merge_requests/1864

to have the image`registry.redhat.io/devspaces/idea-server-rhel8:3.14` published

### What does this PR do?
Adds IDEA desktop editor definition.

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-5758

### CODE REVIEW

File: dependencies/che-plugin-registry/che-editors.yaml [hunk 1]
```diff
@@ -279,3 +279,70 @@ editors:
...
+      name: che-incubator/che-idea-server/latest
+      displayName: IntelliJ IDEA Ultimate (desktop)
+      description: JetBrains IntelliJ IDEA Ultimate dev server for Eclipse Che - latest
+      icon: https://resources.jetbrains.com/storage/products/intellij-idea/img/meta/intellij-idea_logo_300x300.png
+      attributes:
+        publisher: che-incubator
+        version: latest
+        title: JetBrains IntelliJ IDEA Ultimate dev server for Eclipse Che - latest
+        repository: 'https://github.com/che-incubator/che-idea-dev-server'
+        firstPublicationDate: '2023-30-11'
```
[the remark this answers is not in the document]
Comment (azatsarynnyy) [reply dependencies/che-plugin-registry/che-editors.yaml]: updated

File: dependencies/che-plugin-registry/che-editors.yaml [hunk 2]
```diff
@@ -279,3 +279,70 @@ editors:
...
+  - schemaVersion: 2.2.2
+    metadata:
+      name: che-incubator/che-idea-server/latest
+      displayName: IntelliJ IDEA Ultimate (desktop)
+      description: JetBrains IntelliJ IDEA Ultimate dev server for Eclipse Che - latest
+      icon: https://resources.jetbrains.com/storage/products/intellij-idea/img/meta/intellij-idea_logo_300x300.png
+      attributes:
+        publisher: che-incubator
+        version: latest
+        title: JetBrains IntelliJ IDEA Ultimate dev server for Eclipse Che - latest
```
[the remark this answers is not in the document]
Comment (azatsarynnyy) [reply dependencies/che-plugin-registry/che-editors.yaml]: renamed

File: dependencies/che-plugin-registry/che-editors.yaml [hunk 3]
```diff
@@ -279,3 +279,70 @@ editors:
           controller.devfile.io/container-contribution: true
       - name: checode
         volume: {}
+  - schemaVersion: 2.2.2
+    metadata:
+      name: che-incubator/che-idea-server/latest
+      displayName: IntelliJ IDEA Ultimate (desktop)
+      description: JetBrains IntelliJ IDEA Ultimate dev server for Eclipse Che - latest
```
[the remark this answers is not in the document]
Comment (azatsarynnyy) [reply dependencies/che-plugin-registry/che-editors.yaml]: renamed

**Comment by azatsarynnyy** [comment]
Thanks @svor for reviewing the PR! I addressed all your comments.

**Review by svor — APPROVED** [review]
I haven't tested it because the image is not published yet, but changes look good to me.

@azatsarynnyy probably we also need to generate devworkspaces with idea-server editor for our getting started samples, for this need to update devfile registry part: https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/dependencies/che-devfile-registry/build/scripts/generate_devworkspace_templates.sh#L65-L71

**Pull request closed without merging**